capstone-backend/src/main/java/com/google/sps/servlets/SearchServlet.java

```diff
@@ -49,11 +49,11 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     if (searchTerm == null || searchTerm.isEmpty()) return;
     searchTerm = URLEncoder.encode(searchTerm, "UTF-8");
 
-    int maxResults = parseNaturalNumber(request.getParameter("maxResults"));
-    if (request.getParameter("maxResults") != null) { 
-      maxResults = parseNaturalNumber(request.getParameter("maxResults"));
-    } 
-    if (maxResults == -1) {
+    int maxResults = 0;
+    String maxResultsParam = request.getParameter("maxResults"); 
+    if (maxResultsParam != null) { 
+      maxResults = parseNaturalNumber(maxResultsParam);
+    } else if (maxResultsParam == null || maxResults == -1) {
```

Comment:
What happens if `maxResultsParam = `"-1" or some other invalid string (could contain letters, be a negative number, be the empty string, etc)? It would satisfy the first `if` clause (as it is not `null`), but then it would not hit the `else if`, and thus `maxResults` would be set to `-1` instead of `DEFAULT_MAX_RESULTS`. Changing the `else if` to just an `if` clause would fix this problem.

Reply:
Yup! This makes sense. I thought there would be a cleaner way to do it where I'm not doing specific null/not null checks by using an else block but the way I wrote it is not what I intended.

Thank you!